metrics/lib/common.bash

```diff
@@ -42,3 +42,54 @@ function save_results(){
 	bash $LIB_DIR/send_results.sh -n "$1" -a "$2" -r "$3" -u "$4"
 }
 
+# Find a reasonable path to the hypervisor on this system
+function get_qemu_path(){
+    ## Regular expressions used for TOML parsing
+    # Matches a section header
+    section_re="^\s*\[(\S+)]"
+    # Matches the name of the hypervisor section
+    hypervisor_re="^hypervisor\..*"
+    # Matches the variable containing the qemu path
+    qemu_re="^\s*path\s*=\s*\"([^\"]+)"
+
+    # Create a list of potential configuration files
+    declare -a conf_files
+    conf_files+=("${LIB_DIR}/../../../runtime/config/configuration.toml")
+    conf_files+=($(find /etc -type f -name configuration.toml -exec \
+        grep -l 'hypervisor.qemu' {} + 2>/dev/null))
+
+    # Check the potential files sequentially
+    for ((i=0; i<${#conf_files[@]}; i++)); do
```

Comment:
I think this would be a lot clearer if you don't use an index variable, jump out of the loops asap (to avoid the deep indentation) and just use single square brackets. Something like:

```
for conf_file in ${conf_files[*]}; do
    [ -f "${conf_file}" ] || continue

    # ...

done
```